src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -388,43 +399,83 @@ public void onUploadFinishedEVent(final String photoId) {
+        }
+
         oauthRequest.setRequestUrl("https://api.flickr.com/services/rest");
-        oauthRequest.setOAuthToken(configManager.getFlickrToken());
-        oauthRequest.setOAuthTokenSecret(configManager.getFlickrTokenSecret());
         oauthRequest.initParam(OAuthParams.getOAuthParams()
-                .put(OAuthParams.OAUTH_TOKEN, configManager.getFlickrToken())
+                .put(OAuthParams.OAUTH_TOKEN, oauthRequest.getOAuthToken())
                 .put("method", "flickr.photos.addTags")
                 .put("photo_id", photoId)
-                .put("tags", "openstreetmap").toMap());
-        oauthRequest.signRequest(Verb.GET);
-        flickrPhotoClient.setProperties(oauthRequest.getParams(), new Callback<String>() {
-            @Override
-            public void success(String s, Response response) {
-                progressDialog.dismiss();
-                photoFile.delete();
-                Toast.makeText(PhotoActivity.this, "Votre photo a été envoyée. Elle s'affichera dans quelques minutes", Toast.LENGTH_LONG).show();
-                if (ImageAdapter.getPhotoUrlsCachedThumbs(poiId) == null || ImageAdapter.getPhotoUrlsCachedThumbs(poiId).isEmpty()) {
-                    finish();
-                }
-            }
+                .put("tags", flickrOsmTag).toMap());
 
-            @Override
-            public void failure(RetrofitError error) {
-                progressDialog.dismiss();
+        oauthRequest.signRequest(Verb.POST);
+
+        //Filter params for Authorization header
+        Map<String, String> oauthParams = new TreeMap<String, String>();
+        for (Map.Entry<String, String> entry : oauthRequest.getParams().entrySet()) {
+            if (entry.getKey().startsWith("oauth_")) {
+                oauthParams.put(entry.getKey(), entry.getValue());
             }
-        });
+        }
+
+        flickrAddTagClient = FlickrPhotoUtils.getAdapter(oauthParams).create(FlickrAddTagClient.class);
+
+        flickrAddTagClient.addTags(
+                new TypedString("method=flickr.photos.addTags&photo_id=" + photoId + "&tags=" + flickrOsmTag),
+                new Callback<String>() {
+                    @Override
+                    public void success(String s, Response response) {
+                        System.out.println(s);
+
+                        progressDialog.dismiss();
+                        photoFile.delete();
+                        Toast.makeText(PhotoActivity.this, "Votre photo a été envoyée. Elle s'affichera dans quelques minutes", Toast.LENGTH_LONG).show();
+                        if (ImageAdapter.getPhotoUrlsCachedThumbs(poiId) == null || ImageAdapter.getPhotoUrlsCachedThumbs(poiId).isEmpty()) {
+                            finish();
+                        }
+                    }
+
+                    @Override
+                    public void failure(RetrofitError error) {
+                        error.printStackTrace();
```

Comment:
Don't print stacktrace as is. This dirties the android syslog.
Use Debug condition if it is important, otherwise use meaningful android log.